### PR TITLE
solution for #averageAges

### PR DESCRIPTION
**Pull request by jstmpelowycz**

### CODE REVIEW

**Review by SanyaBratashchuk — APPROVED** [review]
good readable code. good job

File: src/averageAges.js [hunk 1]
```diff
@@ -14,12 +14,15 @@
...
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = getPeopleBySex(people, 'm');
+
+  if (century) {
```
Comment (solaryasha) [inline src/averageAges.js:21]: Let's practice ternary and logical operators
```
replace `if ()` statement with &&, || or ?:

File: src/averageAges.js [hunk 2]
```diff
@@ -36,15 +39,24 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = getPeopleBySex(people, 'f');
+
+  if (withChildren) { // O(N^2) :(
```
Comment (solaryasha) [inline src/averageAges.js:46]: Use hash map for avoiding n**2 complexity

You can implement hash map with `reduce` and `{}` as initial value

**Review by solaryasha — CHANGES_REQUESTED** [review]

File: src/averageAges.js (hunk 2 above)
Comment (solaryasha) [reply src/averageAges.js:46]: It's not required, but it will be good practice for you.

**Review by solaryasha — APPROVED** [review]

File: src/averageAges.js (hunk 2 above)
Comment (jstmpelowycz) [reply src/averageAges.js]:
```s

Comment (jstmpelowycz) [reply src/averageAges.js]: ''фф''

Comment (jstmpelowycz) [reply src/averageAges.js]: *и*

**Pull request closed without merging**